blueflood-http/src/main/java/com/rackspacecloud/blueflood/outputs/handlers/HttpMetricDataQueryServer.java

```diff
@@ -79,6 +80,9 @@ public void startServer() {
                         Executors.newFixedThreadPool(workerThreads)));
         server.setPipelineFactory(new MetricsHttpServerPipelineFactory(router));
         serverChannel =  (ServerChannel) server.bind(new InetSocketAddress(httpQueryHost, httpQueryPort));
+
+        log.info("Starting tracker service");
+        Tracker.register();
```

Comment:
If your class is a singleton, it gets instantiated during application startup or when classes are getting loaded depending upon whether you are doing lazy instantiation. Either way, if you move the register call inside constructor, it should work. No need to call the register() method from anywhere.